cmd/ignite/cmd/cmdutil/volumeflag.go

```diff
@@ -0,0 +1,104 @@
+package cmdutil
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/spf13/pflag"
+	api "github.com/weaveworks/ignite/pkg/apis/ignite"
+	"github.com/weaveworks/ignite/pkg/util"
+)
+
+var (
+	formatErr   = fmt.Errorf("volumes must be specified in the /host/path:/vm/path format")
+	absPathsErr = fmt.Errorf("paths given to the volume flag must be absolute")
+)
+
+// VolumeFlag is the pflag.Value custom flag for `ignite create --volume`
+type VolumeFlag struct {
+	value *api.VMStorageSpec
+	s     string
+}
+
+var _ pflag.Value = &VolumeFlag{}
+
+func (vf *VolumeFlag) Set(x string) error {
+	entries := strings.Split(x, ",") // We take in a comma-separated list
+	var storage api.VMStorageSpec
+
+	for i, entry := range entries {
+		paths := strings.Split(entry, ":")
+
+		var hostPath, vmPath string
+		switch len(paths) {
+		case 0:
+			// No paths given
+			continue
+		case 2: // We were given
+			// Set the VM path and verify it's absolute
+			if vmPath = paths[1]; !path.IsAbs(vmPath) {
+				return absPathsErr
```

Comment:
If you like, this can be done here too, but this validation should exist in `pkg/apis/ignite/validation`

Reply:
Will move/add it to `validation`

Reply:
Done in ca75db1.